packages/web/src/pages/comment-history/components/desktop/CommentHistoryPage.tsx

```diff
@@ -0,0 +1,221 @@
+      navigate(track.permalink)
+    }
+  }, [track, navigate])
+
+  if (!comment) return null
+
+  return (
+    <Flex w='100%' gap='l' pv='m'>
+      <Box>
+        {/* @ts-expect-error - the id is a number */}
+        <Avatar userId={userId} size='medium' popover alignSelf='flex-start' />
+      </Box>
+      <Flex column w='100%' gap='s' alignItems='flex-start'>
+        <Flex column gap='xs' w='100%'>
+          <Flex>
+            <Text variant='body' size='s'>
+              {track ? (
+                <>
+                  <TrackLink isActive trackId={track?.track_id} />
+                  {messages.by}
+                  <UserLink isActive userId={track?.owner_id} />
+                </>
+              ) : (
+                <Skeleton w={180} h={20} />
+              )}
+            </Text>
+          </Flex>
+          <Flex column>
+            <Flex justifyContent='space-between'>
+              <Flex gap='s' alignItems='center'>
+                {isLoadingUser ? <Skeleton w={80} h={18} /> : null}
+                {userId !== undefined ? (
+                  <UserLink
+                    // @ts-expect-error - the id is a number
+                    userId={userId}
+                    popover
+                    size='l'
+                    strength='strong'
+                  />
+                ) : null}
+                <Flex gap='xs' alignItems='flex-end' h='100%'>
+                  <Timestamp time={createdAtDate} />
+                </Flex>
+              </Flex>
+            </Flex>
+            <CommentText
+              isEdited={isEdited}
+              isPreview={false}
+              mentions={mentions}
+              // @ts-expect-error - the id is a number
+              commentId={id}
+            >
+              {message}
+            </CommentText>
+          </Flex>
+        </Flex>
+        <Flex gap='l' alignItems='center'>
+          <Flex alignItems='center' gap='xs'>
+            <IconButton
+              icon={IconHeart}
+              color={isCurrentUserReacted ? 'active' : 'subdued'}
+              aria-label='Heart comment'
+              onClick={goToTrackPage}
+            />
+            <Text> {reactCount || ''}</Text>
+          </Flex>
+          <TextLink variant='subdued' onClick={goToTrackPage}>
+            {messages.view}
+          </TextLink>
+        </Flex>
+      </Flex>
+    </Flex>
+  )
+}
+
+export type CommentHistoryPageProps = {
+  title: string
+}
+
+export const CommentHistoryPage = ({ title }: CommentHistoryPageProps) => {
+  const profileParams = useProfileParams()
+  const containerRef = useRef<HTMLDivElement>(null)
+  const { data: currentUserId } = useGetCurrentUserId({})
+  const { data: user } = useGetUserByHandle({
+    handle: profileParams?.handle ?? '',
+    currentUserId
+  })
+
+  const {
+    data: commentIds,
+    hasNextPage,
+    fetchNextPage,
+    isLoading,
+    isFetchingNextPage
+  } = useGetCommentsByUserId({
+    userId: user?.user_id ?? 0,
```

Comment:
Prefer we don't pass 0 here as that could cause bugs. I think we have a convention already for just passing the undefined value and using `enabled` either in the hook or here to disable fetching until we have a userId.

Reply:
the hook checks against 0 in the enabled check, but i'll update to null as that seems more correct